snipsowm/sentence_generator.py

```diff
@@ -201,18 +195,18 @@ def generate_condition_sentence(self,
 
             """
         introduction = self.generate_sentence_introduction(tone)
-
         locality = self.generate_sentence_locality(POI, Locality, Region, Country)
-
         date = self.generate_sentence_date(date, granularity=granularity)
 
-        permutable_parameters = list((locality, date))
+        permutable_parameters = [locality, date]
         random.shuffle(permutable_parameters)
-        parameters = (introduction, condition_description) + tuple(permutable_parameters)
+        parameters = [introduction, condition_description] + permutable_parameters
 
         # Formatting
-        parameters = filter(lambda x: not x is None and len(x) > 0, parameters)
-        return ("{} " * len(parameters)).format(*parameters)
+        parameters = [x for x in parameters if x]
+        # TODO was this space in the end intended?
+        return ' '.join(parameters) + ' '
```

Comment:
is it intended that there is a space in the end? I just added this because that would be the result of the old code, but it does not seem like this behaviour is intended